alternator/executor.cc

```diff
@@ -2954,24 +2979,60 @@ future<executor::request_return_type> executor::batch_write_item(client_state& c
                 co_return api_error::validation(fmt::format("Unknown BatchWriteItem request type: {}", r_name));
             }
         }
-        per_table_wcu.emplace_back(per_table_stats, wcu_delete_units, wcu_put_units);
-        if (should_add_wcu) {
-            rjson::value entry = rjson::empty_object();
-            rjson::add(entry, "TableName", rjson::from_string(rjson::to_string_view(it->name)));
-            rjson::add(entry, "CapacityUnits", wcu_delete_units + wcu_put_units);
-            rjson::push_back(consumed_capacity, std::move(entry));
-        }
+        per_table_wcu.emplace_back(per_table_stats, schema, rjson::to_string_view(it->name));
     }
     for (const auto& b : mutation_builders) {
         co_await verify_permission(_enforce_authorization, client_state, b.first, auth::permission::MODIFY);
     }
-    wcu_put_units = 0;
-    wcu_delete_units = 0;
+
+    if (needs_read_before_write) {
```

Comment:
If I understand correctly, this is the only place in this function you need needs_read_before_write(), and you defined in in the previous patch to be equal to that force_read_before_write configuration variable. I think you should move this definition here, or even better - don't have this flag at all, just have a
```
if (config.force_read_before_write) {
```
And then it will be more obvious that all this extra code only affects the rare force_read_before_write case.